src/main/java/refinedstorage/tile/TileDestructor.java

```diff
@@ -71,9 +95,29 @@ public int getEnergyUsage() {
     @Override
     public void updateNode() {
         if (ticks % upgrades.getSpeed(BASE_SPEED, 4) == 0) {
-            if (type == IType.ITEMS) {
-                BlockPos front = pos.offset(getDirection());
 
+            BlockPos front = pos.offset(getDirection());
+
+            if (pickupItem && type == IType.ITEMS) {
+                List<Entity> droppedItem = new ArrayList<>();
+
+                Chunk chunk = worldObj.getChunkFromBlockCoords(front);
+                chunk.getEntitiesWithinAABBForEntity(null, new AxisAlignedBB(front), droppedItem, null);
+
+                if (droppedItem.size() > 0) {
+                    for (Entity item : droppedItem) {
+                        if (item instanceof EntityItem) {
+                            ItemStack remainder = network.insertItem(((EntityItem) item).getEntityItem().copy(), ((EntityItem) item).getEntityItem().stackSize, false);
+                            if (remainder != null) {
+                                InventoryHelper.spawnItemStack(worldObj, front.getX(), front.getY(), front.getZ(), remainder);
```

Comment:
don't insert and then respawn an item when there is no space

just simulate an  insert first and check if the insert was successful before really inserting and removing the item